### PR TITLE
Move distance snap grid to osu! HitObjectComposer

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on #9070

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs [hunk 1]
```diff
@@ -32,9 +37,66 @@ protected override DrawableRuleset<OsuHitObject> CreateDrawableRuleset(Ruleset r
...
+        {
+            if (distanceSnapGrid == null)
+                return base.SnapScreenSpacePositionToValidTime(screenSpacePosition);
+
+            (Vector2 pos, double time) = distanceSnapGrid.GetSnappedPosition(distanceSnapGrid.ToLocalSpace(screenSpacePosition));
+
+            return new SnapResult(distanceSnapGrid.ToScreenSpace(pos), time);
+        }
+
+        private void updateDistanceSnapGrid()
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs:84]: This also needs to be called when the select tool is selected. Without that the placement grid remains visible.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs:84]: Aha. I was only testing using 1 / 2 hotkeys, where it works as expected. Not 100% happy on the invalidation logic for this

Comment (peppy) [reply osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs:84]: Fix in a better feeling way – now the grid will disappear when exiting the placement area.

**Review by smoogipoo — APPROVED** [review]
Seems to work fine for me

**Pull request merged**